src/PostageStamp.sol

```diff
@@ -138,32 +177,34 @@ contract PostageStamp is AccessControl, Pausable {
         bytes32 _nonce,
         bool _immutable
     ) external whenNotPaused {
-        require(_owner != address(0), "owner cannot be the zero address");
-        // bucket depth should be non-zero and smaller than the depth
-        require(
-            _bucketDepth != 0 && minimumBucketDepth <= _bucketDepth && _bucketDepth < _depth,
-            "invalid bucket depth"
-        );
-        // derive batchId from msg.sender to ensure another party cannot use the same batch id and frontrun us.
+        if (_owner == address(0)) {
+            revert ZeroAddress();
+        }
+
+        if (_bucketDepth == 0 || _bucketDepth < minimumBucketDepth || _bucketDepth >= _depth) {
+            revert InvalidDepth();
+        }
+
         bytes32 batchId = keccak256(abi.encode(msg.sender, _nonce));
-        require(batches[batchId].owner == address(0), "batch already exists");
-        require(
-            _initialBalancePerChunk >= minimumInitialBalancePerChunk(),
-            "insufficient initial balance for 24h minimum validity"
-        );
-        // per chunk balance multiplied by the batch size in chunks must be transferred from the sender
+        if (batches[batchId].owner != address(0)) {
+            revert BatchExists();
+        }
+
+        if (_initialBalancePerChunk < minimumInitialBalancePerChunk()) {
+            revert InsufficientBalance();
+        }
+
         uint256 totalAmount = _initialBalancePerChunk * (1 << _depth);
-        require(ERC20(bzzToken).transferFrom(msg.sender, address(this), totalAmount), "failed transfer");
+        if (!ERC20(bzzToken).transferFrom(msg.sender, address(this), totalAmount)) {
+            revert TransferFailed();
+        }
 
-        // normalisedBalance is an absolute value per chunk, as if the batch had existed
-        // since the block the contract was deployed, so we must supplement this batch's
-        // _initialBalancePerChunk with the currentTotalOutPayment()
         uint256 normalisedBalance = currentTotalOutPayment() + (_initialBalancePerChunk);
+        if (normalisedBalance <= 0) {
```

Comment:
can it be even below zero ever? since it is an uint.

Reply:
yes u r right.  Just did it auto opposite of >  but it can be just == 
Fixed